src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -236,14 +236,22 @@ private fun checkDot3Parameter(holder: AnnotationHolder, dot3: PsiElement?) {
 
 private fun checkValueParameter(holder: AnnotationHolder, param: RsValueParameter) {
     val fn = param.parent.parent as? RsFunction ?: return
+    val isPattern = param.pat != null && (param.pat as? RsPatIdent) == null
```

Comment:
This condition seems wrong since `rustc` allows at least a wild pattern:
```rust
trait Foo {
    fn foo(_: i32);
}
```

Please double check the logic with the rust compiler.

Also, let's extract `param.pat` to a variable since it is used later.
Also, let's rename `isPattern` to a something like `isComplexPattern` since strictly saying  `let a;` is a pattern as well.